### DISCUSSION TITLE
Tutorial app for Serverless functions throws Error

### DISCUSSION THREAD

**Discussion by teohaik** [vercel/community#2695] [Help] [answered]
### Summary

Hello,

iam trying to follow [this guide](https://vercel.com/docs/concepts/functions/serverless-functions/quickstart) for a simple serverless functions but there is discrepancy between the tutorial and the code created by the command: `npx create-next-app@latest --typescript`

The tutorial presumes that folder pages/api/ exists. but it doesn't. 
Also the scaffolded app contains an src folder.

I manually created the folders and put the handler.ts file there, but iam getting an error 

![image](https://github.com/vercel/community/assets/1785012/1ea73a84-6d54-47e7-8539-79bb5889fa20)

### Steps to Reproduce

Run 
`npx create-next-app@latest --typescript`

create `src/pages/api` folder and handlers.ts file with:

```
import { NextApiRequest, NextApiResponse } from 'next';
 
export default function handler(
  request: NextApiRequest,
  response: NextApiResponse,
) {
  response.status(200).json({
    body: request.body,
    query: request.query,
    cookies: request.cookies,
  });
}
```

run `vercel dev`

**Comment by amyegan** [comment, marked as answer]
Hi @teohaik. It looks like that tutorial was written for the older Pages Router structure rather than the new App Router. Thanks for pointing that out so we can get it updated! 

For help setting up serverless functions in the `app` directory, you can use the [Next.js Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/router-handlers) documentation.

**Reply by leerob** [reply]
To further clarify, this is okay:

```ts
import { NextApiRequest, NextApiResponse } from 'next';
 
export default function handler(
  request: NextApiRequest,
  response: NextApiResponse,
) {
  response.status(200).json({
    body: request.body,
    query: request.query,
    cookies: request.cookies,
  });
}
```

But since you added:

```ts
export const config = {
  runtime: 'edge',
};
```

It will not work. It needs to be:

```ts
import { NextRequest, NextResponse } from 'next/server';
 
export const config = {
  runtime: 'edge',
};
 
export default (request: NextRequest) => {
  return NextResponse.json({
    name: `Hello, from ${request.url} I'm now an Edge Function!`,
  });
};
```

if you are using the Pages Router, or Route Handlers as mentioned about the App Router, which allows you to more easily move between runtimes without changing the function interface.
